odl/util/numerics.py

```diff
@@ -297,6 +303,521 @@ def fast_1d_tensor_mult(ndarr, onedim_arrs, axes=None, out=None):
     return out
 
 
+def resize_array(arr, newshp, frac_left=None, num_left=None,
+                 pad_mode='constant', pad_const=0, out=None):
+    """Return the resized version of ``arr`` with shape ``newshp``.
+
+    In axes where ``newshp > arr.shape``, padding is applied according
+    to the supplied options.
+    Where ``newshp < arr.shape``, the array is cropped to the new
+    size.
+
+    Parameters
+    ----------
+    arr : array-like
+        Array to be resized.
+    newshp : sequence of int
+        Desired shape of the output array.
+    frac_left : float or sequence of float, optional
+        Number between 0 and 1 (inclusive) that determines which fraction
+        of the addition/removal is performed on the "left" side. If
+        ``frac_left`` times the shape difference is not integer, the
+        number of entries added to the left is gained by rounding to the
+        nearest integer. The default ``None`` is equivalent to 0.5.
+        This option can be specified per axis with a sequence.
+        Cannot be combined with ``num_left``.
```

Comment:
Actually, if I think about it, this option should be removed. It's not super reliable due to rounding, and the same effect can easily be reached by a simple calculation by the user. Opinions?